web/source/osk/visualKeyboard.ts

```diff
@@ -1072,29 +1107,72 @@ namespace com.keyman.osk {
         paddedHeight = this.computedAdjustedOskHeight(this.height);
       }
 
-      let b = this.kbdDiv.firstChild as HTMLElement;
+      let b = this.layerGroup.element as HTMLElement;
       let gs = this.kbdDiv.style;
       let bs=b.style;
       if(this.usesFixedHeightScaling) {
         // Sets the layer group to the correct height.
         gs.height = gs.maxHeight = paddedHeight + 'px';
       }
-      bs.fontSize=fs+'em';
+
+      // The font-scaling applied on the layer group.
+      gs.fontSize = this.fontSize.styleString;
+      bs.fontSize=ParsedLengthStyle.forScalar(fs).styleString;
 
       // Needs the refreshed layout info to work correctly.
       for(const layerId in this.layerGroup.layers) {
         const layer = this.layerGroup.layers[layerId];
         layer.refreshLayout(this, paddedHeight, this.height);
       }
+
+      // NEW CODE ------
```

Comment:
Let's not have comments like this in the final work -- it's only NEW CODE today. An appropriate comment would be to ref the overarching issue #, e.g. `// #1234 - version 15 layout system`